### PR TITLE
codeintel: Add additional usage telemetry

### PR DESCRIPTION
**Pull request by efritz**
Updates the payload sent by server update checks to have the following new fields for code intelligence usage.

```json
"new_codeintel_usage": {
    "start_time": "...",
    "waus": 100,
    "precise_waus": 100,
    "search_waus": 100,
    "xrepo_waus": 100,
    "precise_xrepo_waus": 100,
    "search_xrepo_waus": 100,
    "event_summaries": [
        {
            "action": "...",
            "source": "...",
            "language_id": "...",
            "cross_repository": "...",
            "waus": "...",
            "total_actions": "..."
        },
        ...
    ],
    "num_repositories_with_upload_records": 100,
    "num_repositories_without_upload_records": 100,

    // New fields
    "num_repositories": 100,
    "num_repositories_with_fresh_upload_records": 100,
    "num_repositories_with_index_records": 100,
    "num_repositories_with_fresh_index_records": 100,
    "num_repositories_with_index_configuration_records": 100,
    "counts_by_language": [
        {
            "language_id": "go",
            "num_repositories_with_upload_records": 100,
            "num_repositories_with_fresh_upload_records": 100,
            "num_repositories_with_index_records": 100,
            "num_repositories_with_fresh_index_records": 100,
        },
        {
            "language_id": "typescript",
            "num_repositories_with_upload_records": 100,
            "num_repositories_with_fresh_upload_records": 100,
            "num_repositories_with_index_records": 100,
            "num_repositories_with_fresh_index_records": 100
        },
        ...
    },
    "settings_page_view_count": 100
}
```

### CODE REVIEW

**Review by macraig — APPROVED** [review]
@kelsey-brown can you please review on behalf of bizops to make sure we're all good with the data collected?

**Comment by kelsey-brown** [comment]
@attfarhan I'm out of my league on this one - can you take a look at some point? The code intel team wants to be able to query repo language (we can discuss in our 1-1 as well if you want some additional context)

**Review by attfarhan — APPROVED** [review]
Approving on behalf of BizOps for the new data points being collected (@kelsey-brown it sounds like you all have been working through this here as well https://github.com/sourcegraph/analytics/issues/232). But please don't merge until the schema is updated!

Did not do a full code review, leaving that for the dev team.

@efritz I can also review your schema update PR once ready and test it so we can get everything merged.

**Comment by efritz** [comment]
@attfarhan Schema PR here: https://github.com/sourcegraph/analytics/pull/267. Could you review and apply?

File: cmd/frontend/internal/app/updatecheck/handler.go [hunk 1]
```diff
@@ -420,17 +420,46 @@ func reserializeNewCodeIntelUsage(payload json.RawMessage) (json.RawMessage, err
...
+		PreciseCrossRepositoryWAUs:                   codeIntelUsage.PreciseCrossRepositoryWAUs,
+		SearchBasedCrossRepositoryWAUs:               codeIntelUsage.SearchBasedCrossRepositoryWAUs,
+		EventSummaries:                               eventSummaries,
+		NumRepositories:                              numRepositories,
+		NumRepositoriesWithUploadRecords:             codeIntelUsage.NumRepositoriesWithUploadRecords,
+		NumRepositoriesWithoutUploadRecords:          numRepositoriesWithoutUploadRecords,
+		NumRepositoriesWithFreshUploadRecords:        codeIntelUsage.NumRepositoriesWithFreshUploadRecords,
+		NumRepositoriesWithIndexRecords:              codeIntelUsage.NumRepositoriesWithIndexRecords,
+		NumRepositoriesWithFreshIndexRecords:         codeIntelUsage.NumRepositoriesWithFreshIndexRecords,
+		NumRepositoriesWithIndexConfigurationRecords: codeIntelUsage.NumRepositoriesWithAutoIndexConfigurationRecords,
```
Comment (coury-clark) [inline cmd/frontend/internal/app/updatecheck/handler.go:460]: Just to make sure, the different names on this row are intentional?

Comment (efritz) [reply cmd/frontend/internal/app/updatecheck/handler.go:460]: 🙏 

Reviews are nice

Comment (efritz) [reply cmd/frontend/internal/app/updatecheck/handler.go:460]: Yes the last two actually do match, but I had to make sure 😓 . Thanks!

**Review by coury-clark — APPROVED** [review]

**Review by unknwon — APPROVED** [review]

**Comment by efritz** [comment]
@attfarhan Please check the PR description (updated) against the BigQuery schema. This should be the same shape as the test value in https://github.com/sourcegraph/sourcegraph/pull/23587/files#diff-15a105fe37bbb999549afc484c3f9e459615e1a5659d1fa1028452c9f7c9f52cR526.

**Comment by attfarhan** [comment]
@efritz the schema has been updated, feel free to merge.

**Comment by macraig** [comment]
@kelsey-brown we're now ready to start adding to the dashboard

**Pull request merged**